src/Storages/MergeTree/MergeTreeIndexText.cpp

```diff
@@ -775,6 +776,27 @@ MergeTreeIndexGranulePtr MergeTreeIndexAggregatorText::getGranuleAndReset()
     return granule;
 }
 
+namespace
+{
+void addArrayValuesIntoGranuleBuilder(
```

Comment:
Minor: I guess we don't need to have a separate function for arrays. It is not called from >1 places and the other path (l. 822ff) is also not in a separate function.